### PR TITLE
feature: add pipeline and actionagent

### PR DESCRIPTION
**Pull request by sfwn**
#### What type of PR is this?

feature

#### What this PR does / why we need it:

add pipeline & actionagent to erda

### CODE REVIEW

File: modules/actionagent/agenttool/tar.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import (
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/erda-project/erda/pkg/retry"
+
+	"github.com/mholt/archiver"
```
Comment (Effet) [inline modules/actionagent/agenttool/tar.go:12]: bad order

**Review by Effet — CHANGES_REQUESTED** [review]

File: modules/actionagent/agenttool/tar.go (hunk 1 above)
Comment (sfwn) [reply modules/actionagent/agenttool/tar.go]: adjusted.

**Review by Effet — APPROVED** [review]
a horrible huge PR!

**Pull request merged**